Index: source/parser/plugins/CssAtKeyframesParserPlugin.php
===================================================================
--- source/parser/plugins/CssAtKeyframesParserPlugin.php	(revision 167)
+++ source/parser/plugins/CssAtKeyframesParserPlugin.php	(working copy)
@@ -65,6 +65,22 @@
 			$this->parser->clearBuffer();
 			return $index + 15;
 			}
+		// Start of @keyframes at-rule block (@-ms-keyframes)
+		else if ($char === "@" && $state === "T_DOCUMENT" && strtolower(substr($this->parser->getSource(), $index, 14)) === "@-ms-keyframes")
+		        {
+			$this->atRuleName = "-ms-keyframes";
+			$this->parser->pushState("T_AT_KEYFRAMES::NAME");
+			$this->parser->clearBuffer();
+			return $index + 14;
+			}
+		// Start of @keyframes at-rule block (@-o-keyframes)
+		else if ($char === "@" && $state === "T_DOCUMENT" && strtolower(substr($this->parser->getSource(), $index, 13)) === "@-o-keyframes")
+			{
+			$this->atRuleName = "-o-keyframes";
+			$this->parser->pushState("T_AT_KEYFRAMES::NAME");
+			$this->parser->clearBuffer();
+			return $index + 13;
+			}
 		// Start of @keyframes at-rule block (@-webkit-keyframes)
 		elseif ($char === "@" && $state === "T_DOCUMENT" && strtolower(substr($this->parser->getSource(), $index, 18)) === "@-webkit-keyframes")
 			{
@@ -158,4 +174,4 @@
 		return true;
 		}
 	}
-?>
\ No newline at end of file
+?>
